### PR TITLE
add firebase

### PR DESCRIPTION
**Pull request by bmorelli25**

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -2,25 +2,81 @@ import type { NextPage } from "next";
...
+      .then((data) => {
+        data.forEach((doc) => {
+          let id = doc.id;
+          let rd = doc.data();
+          rd.id = id;
+          docs.push(rd);
+        })
+        setRoomList(docs)
+      })
+  };
```
Comment (bmorelli25) [inline pages/index.tsx:30]: This function queries FB and returns an array of active rooms. It then sets the `roomList` state to match FB.

File: pages/index.tsx [hunk 2]
```diff
@@ -2,25 +2,81 @@ import type { NextPage } from "next";
...
+    const docID = "randomID"+ currentRoom
+    setDoc(doc(database, "rooms", docID), {
+      room: currentRoom,
+      users: []
+    }).then(() => {
+      setNewRoom('');
+    }).then(() => {
+      router.push(`/room/${newRoom}?roomid=${docID}`);
+    })
+  }
```
Comment (bmorelli25) [inline pages/index.tsx:43]: This function sends a new document to FB. I'm quirkily creating the doc ID right now. We should set this to a random string in the future. After the room is added to FB, it forwards the user to the room page.

File: pages/index.tsx [hunk 3]
```diff
@@ -2,25 +2,81 @@ import type { NextPage } from "next";
...
+        console.log("Error getting documents: ", error);
+    })
+    .then(() => {
+      if (matches.length > 0){
+        router.push(`/room/${joinRoom}/?roomid=${matches[0].id}`)
+      } else {
+        window.alert("room does not exist");
+      }
+    });
+  }
```
Comment (bmorelli25) [inline pages/index.tsx:78]: This function determines if the user supplied room input is an active room. It queries FB looking for a document where `room: joinRoom`. If there's no match, the `matches` array will be empty and for now an alert pops up that says the room does not exist. If there is a match, the rooms array will include one object. We take the ID of that object and add it as a query param then send the user to the room page.

File: pages/room/[id].tsx [hunk 4]
```diff
@@ -1,15 +1,44 @@
...
-  const { id } = router.query
+  const { id, roomid } = router.query
+
+  const deleteRoom = () => {
+    let room = roomid?.toString()
+    deleteDoc(doc(database, "rooms", room))
+      .then(() => {
+        router.push(`/`)
+      })
+  }
```
Comment (bmorelli25) [inline pages/room/[id].tsx:28]: The delete function pulls the FB doc ID out of the query param `roomid`. It sends this roomID to FB to delete the room then forwards the user to the index page.

**Comment by bmorelli25** [comment]
We can probably move all of these functions to a helper file or turn them into API endpoints to clean up the code. I wasn't sure what the best approach is.

**Comment by dstrizzi** [comment]
I created a `utils` folder in my latest PR. We can put all those functions in there.

**Review by dstrizzi — APPROVED** [review]
WOWOWOW, this is really great. Who would have guessed that FireBase would be the way to go....

We can clean it up in a separate PR if you just want to merge this for now.

**Pull request merged**